dataRetriever/resolvers/headerResolver.go

```diff
@@ -113,18 +119,31 @@ func (hdrRes *HeaderResolver) ProcessReceivedMessage(message p2p.MessageP2P, _ f
 	return hdrRes.Send(buff, message.Peer())
 }
 
-func (hdrRes *HeaderResolver) resolveHeaderFromNonce(key []byte) ([]byte, error) {
+func (hdrRes *HeaderResolver) resolveHeaderFromNonce(rd *dataRetriever.RequestData) ([]byte, error) {
 	// key is now an encoded nonce (uint64)
 
 	// Search the nonce-key pair in cache-storage
-	hash, err := hdrRes.hdrNoncesStorage.Get(key)
+	hash, err := hdrRes.hdrNoncesStorage.GetFromEpoch(rd.Value, rd.Epoch)
```

Comment:
There is no need to search in a wrong epoch, if you now it better where it should be searched.

Reply:
extracted the cache search into a function, but left the storage search as it is because I need the nonce and the epoch also in the function and therefore if I would create a function for storage search it would have only one line